.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+    branches: ["main"]
+  workflow_call:
+
+
+# In cases of concurrent workflows running (consecutive pushes to PR)
+# leave the latest workflow and cancel the other (older) workflows
+# See https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  actions: write
+  contents: read
+
+
+name: UI Bundle CI
+
+
+jobs: 
+  ui_bundle_skip:
+    name: UI Bundle Skip Unchanged
+    continue-on-error: true # Uncomment once integration is finished
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+      run_id: ${{ steps.skip_check.outputs.skipped_by.id }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v5
+        with:
+          concurrent_skipping: "never"
+          skip_after_successful_duplicate: "true"
+          do_not_skip: '["workflow_dispatch", "schedule"]'
+
+  ui_bundle_build: 
+   name: UI Bundle Build 
+   runs-on: ubuntu-latest
+   needs: [ui_bundle_skip]
+   if: needs.ui_bundle_skip.outputs.shjould_skip != 'true'
+   steps: 
+    - uses: actions/checkout@v3
+    - name: Install Node 
+      uses: actions/setup-node@v3
+      with:
+       node-version: 16
+
+    - name: Install node dependencies
+      run: npm install --quiet --no-progress && node_modules/.bin/patch-package
+    
+    - name: Lint 
+      run: node_modules/.bin/gulp lint
+
+    - name: Bundle
+      run: node_modules/.bin/gulp bundle
+
+
+    - name: Upload Bundle to Artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: ui-bundle
+        retention-days: 14
+        path: |
+          build/ui-bundle.zip
```

Comment:
Does the latest `main` bundle stay around, or should we have a cronjob build this every ~7 days?

Reply:
My plan is to make a release and have the antora instance rely on the releases. This is for PRs for now and getting something out there I can test.